Cogs/Emoji.py

```diff
@@ -34,6 +34,29 @@ def _get_emoji_mention(self, emoji):
     async def addemoji(self, ctx, *, emoji = None, name = None):
         '''Adds the passed emoji, url, or attachment as a custom emoji with the passed name (bot-admin only, max of 10).'''
         if not await Utils.is_bot_admin_reply(ctx): return
+        # Adds an emoji in a referenced message
+        if not len(ctx.message.attachments) and name == None and ctx.message.reference is not None:

```

Comment:
Checking that a value `== None` is bad practice. Use the `is` keyword.

Reply:
You could also just do `not name`, not sure why you _need_ to check if it's [equal to] `None` at all. This can be shortened down to:

```py
    async def addemoji(self, ctx, *, emoji = None, name = None):
        '''Adds the passed emoji, url, or attachment as a custom emoji with the passed name (bot-admin only, max of 10).'''
        if not await Utils.is_bot_admin_reply(ctx): return

        orig_msg = ctx.message

        # Adds an emoji in a referenced message
        if (
           not len(orig_msg.attachments) and 
           not name and 
           orig_msg.reference
        ):
```